api/handler/acl.go

```diff
@@ -0,0 +1,1149 @@
+package handler
+
+import (
+	"crypto/ecdsa"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neofs-api-go/pkg/acl/eacl"
+	"github.com/nspcc-dev/neofs-api-go/pkg/object"
+	"github.com/nspcc-dev/neofs-s3-gw/api"
+	"github.com/nspcc-dev/neofs-s3-gw/api/layer"
+)
+
+var (
+	writeOps = []eacl.Operation{eacl.OperationPut, eacl.OperationDelete}
+	readOps  = []eacl.Operation{eacl.OperationGet, eacl.OperationHead,
+		eacl.OperationSearch, eacl.OperationRange, eacl.OperationRangeHash}
+	fullOps = []eacl.Operation{eacl.OperationGet, eacl.OperationHead, eacl.OperationPut,
+		eacl.OperationDelete, eacl.OperationSearch, eacl.OperationRange, eacl.OperationRangeHash}
+)
+
+var actionToOpMap = map[string][]eacl.Operation{
+	"s3:DeleteObject": {eacl.OperationDelete},
+	"s3:GetObject":    readOps,
+	"s3:PutObject":    {eacl.OperationPut},
+	"s3:ListBucket":   readOps,
```

Comment:
Is this mapping suppose to be _precise_? For example to control `s3:GetObject` you don't need to control `eacl.OperationSearch`. For `s3:ListBucket` it seems that `eacl.OperationSearch` should be enough (I don't see how one can abuse listing with HEAD, GET, etc. requests).

Larger scopes are not bad, but maybe they can interfere each other (restrict get + allow search = ?).

Reply:
> Is this mapping suppose to be _precise_? For example to control `s3:GetObject` you don't need to control `eacl.OperationSearch`. For `s3:ListBucket` it seems that `eacl.OperationSearch` should be enough (I don't see how one can abuse listing with HEAD, GET, etc. requests).
> 
> Larger scopes are not bad, but maybe they can interfere each other (restrict get + allow listing = ?).

I think we can reconsider this mapping when we add cache to s3-gw. For example at the moment `GetObject` has to perform `search`, `get`, `head`, `range`(sometimes).